src/server/license/server/api_server.go

```diff
@@ -230,9 +230,16 @@ func (a *apiServer) AddCluster(ctx context.Context, req *lc.AddClusterRequest) (
 	}, nil
 }
 
+func stripSecretFromRequest(req *lc.HeartbeatRequest) *lc.HeartbeatRequest {
+	r := *req
```

Comment:
based on the code in the Heartbeat method below, I'm making the assumption that a nil check on the request isn't necessary. Does that seem right @actgardner ?

Reply:
We don't conventionally do it, although it's not a great habit because a malformed request could cause the server to restart. We should probably just add a GRPC middleware to require non-nil requests everywhere.